### ISSUE TITLE
Database schema / install file

### ISSUE THREAD

**Issue by laryn** [backdrop-contrib/date_repeat#18]
I scanned things over today and it seems that perhaps we need to add an install file that adds something to the date field schema to allow the `rrule` to be stored:

 - https://git.drupalcode.org/project/date/-/blob/7.x-2.x/date.install#L89-96

**Comment by laryn** [comment]
For the next time I'm able to look at this, here's a stream in Zulip related:

 - https://backdrop.zulipchat.com/#narrow/stream/218635-Backdrop/topic/Date.20Repeat/near/248976522

**Comment by indigoxela** [comment]
> I scanned things over today and it seems that perhaps we need to add an install file that adds something to the date field schema to allow the rrule to be stored:

Yes, that database column is essential to get the module working properly. @laryn it might happen that I come with some PRs in near future, I hope that's OK for you. (I'm not 100% sure yet, but might need this functionality in a project.)

**Comment by laryn** [comment]
> it might happen that I come with some PRs in near future, I hope that's OK for you.

If such a thing were to happen, I would try my best not to be horribly angry. (Kidding of course, I would welcome a PR!)

**Comment by laryn** [comment]
STATUS: the module port has stopped upon coming to the conclusion that the way that the module previously hacked the database schema will not work in Backdrop. So as @indigoxela summarized, "The field values get saved properly, but the repeat rule is lost."

If someone comes up with an alternative PR to do this I will be happy to review but don't plan to work on it myself. Instead I'll plan to use the new module from @indigoxela: https://github.com/backdrop-contrib/repeating_dates

**Comment by herbdool** [comment]
I'm making some progress on this. I used this core PR https://github.com/backdrop/backdrop-issues/issues/6208 to get it mostly working. It might work without that PR if using `hook_schema_alter()` but I stopped trying to figure out why I couldn't get it to work. This PR gets us most of the way there.

The reason I would still like this module working is to make it easier to support an upgrade of a site that uses Calendar module. In theory, a separate module is possible, such as `repeating_date`, but would need to add full support for Views that core Date provides.
